crates/uv-python/src/discovery.rs

```diff
@@ -977,13 +983,34 @@ pub(crate) fn find_python_installation(
 ) -> Result<FindPythonResult, Error> {
     let installations = find_python_installations(request, environments, preference, cache);
     let mut first_prerelease = None;
+    let mut first_error = None;
+    let mut show_first_error = true;
     for result in installations {
-        // Iterate until the first critical error or happy result
+        // Skip non-critical errors
         if !result.as_ref().err().map_or(true, Error::is_critical) {
+            // Track the first non-critical error
+            if let Err(err) = result {
+                if let Some(ref first) = first_error {
+                    if show_first_error {
+                        debug_show_skipped(first);
+                        show_first_error = false;
+                    }
+                    debug_show_skipped(&err);
```

Comment:
This is a little annoying... but basically we don't want to show the "skipped" log for the first interpreter if we're going to immediately throw it as an error; but we want to show it in the right position.

Reply:
It kind of seems ok to me to log it, then throw it as an error, if it makes the code simpler. I'm having some trouble with it as-is. Like, how does this guard against "first error is non-critical, don't know", then "second error is non-critical; show first and second", then we run out of Pythons, exit the loop, and show the first error as an error (again)?

Reply:
I'll revert it.

Reply:
It's ok if you want to keep it, just feedback as a reader.

Reply:
Naw I was actually confused by the output of `grep` in the integration test and it's not as bad as I thought.

Reply:
It was hard to write too.